Source/WebCore/rendering/style/BorderValue.cpp

```diff
@@ -25,23 +25,26 @@
 #include "config.h"
 #include "BorderValue.h"
 
+#include "RenderStyle.h"
+
 namespace WebCore {
 
 BorderValue::BorderValue()
-    : m_style(static_cast<unsigned>(BorderStyle::None))
+    : m_color(CurrentColor::make())
+    , m_style(static_cast<unsigned>(BorderStyle::None))
     , m_isAuto(static_cast<unsigned>(OutlineIsAuto::Off))
 {
 }
 
-
-bool BorderValue::isTransparent() const
+bool BorderValue::isTransparent(const RenderStyle* renderStyle) const
 {
-    return m_color.isValid() && !m_color.isVisible();
+    auto color = renderStyle ? renderStyle->colorResolvingCurrentColor(m_color) : m_color.resolveColor();
+    return color.isValid() && !color.isVisible();
 }
 
-bool BorderValue::isVisible() const
+bool BorderValue::isVisible(const RenderStyle* renderStyle) const
 {
-    return nonZero() && !isTransparent() && style() != BorderStyle::Hidden;
+    return nonZero() && !isTransparent(renderStyle) && style() != BorderStyle::Hidden;
 }
```

Comment:
Let's not do this (and the large associated refactoring) in this patch. Passing RenderStyle like this is not nice and you again run into question of when is it ok not to pass it.  

You can just use absoluteOrInvalidColorIgnoringCurrentColor() mentioned elsewhere.

This presumable fixes a bug so there should be associated test too.